### PR TITLE
feat: enhance timeFilter/diffSync support for jenkins build

### PR DESCRIPTION
**Pull request by abeizn**
### Summary
feat: enhance timeFilter/diffSync support for jenkins build

### Does this close any open issues?
part of https://github.com/apache/incubator-devlake/issues/4535

### Screenshots
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: backend/plugins/jenkins/tasks/build_collector.go [hunk 1]
```diff
@@ -43,10 +49,15 @@ type SimpleJob struct {
 	Path string
 }
 
+type SimpleJenkinsApiBuild struct {
+	Number    int64
+	Timestamp int64
+}
```
Comment (warren830) [inline backend/plugins/jenkins/tasks/build_collector.go:55]: what is this for? No json tag?

File: backend/plugins/jenkins/tasks/build_collector.go [hunk 2]
```diff
@@ -55,29 +66,68 @@ func CollectApiBuilds(taskCtx plugin.SubTaskContext) errors.Error {
...
+					}
+					err := helper.UnmarshalResponse(res, &data)
+					if err != nil {
+						return nil, err
+					}
+					return data.Builds, nil
+				},
+			},
+			GetCreated: func(item json.RawMessage) (time.Time, errors.Error) {
+				b := &SimpleJenkinsApiBuild{}
```
Comment (warren830) [inline backend/plugins/jenkins/tasks/build_collector.go:96]: i think you should add json tag for SimpleJenkinsApiBuild, Then you can get its'createdDate

File: backend/plugins/jenkins/tasks/build_collector.go (hunk 1 above)
Comment (abeizn) [reply backend/plugins/jenkins/tasks/build_collector.go:54]: done

**Review by mindlesscloud — APPROVED** [review]
LGTM

**Pull request merged**